features/device-config/shell/src/main/java/org/opennms/features/deviceconfig/shell/GetDeviceConfig.java

```diff
@@ -78,13 +84,20 @@ public Object execute() throws Exception {
             System.out.printf("Not a valid host %s \n", host);
             return null;
         }
-        CompletableFuture<byte[]> future = deviceConfigService.getDeviceConfig(host, location, configType, timeout);
+        CompletableFuture<DeviceConfig> future = deviceConfigService.getDeviceConfig(host, location, configType, timeout);
         while (true) {
             try {
                 try {
-                    byte[] dc = future.get(1, TimeUnit.SECONDS);
-                    if (dc != null) {
-                        return new String(dc, Charset.forName(encoding));
+                    DeviceConfig deviceConfig = future.get(1, TimeUnit.SECONDS);
+                    if (deviceConfig != null) {
+                        System.out.printf("Received file %s with content .. \n\n", deviceConfig.getFilename());
+                        if (deviceConfig.getFilename().contains(".gz")) {
```

Comment:
Use `endsWith` instead? Or maybe even use something like real MIME-type detection?

Reply:
Tried MIME-type detection on `.gz`,  `.gz` file parses as OctetStream which couldn't be any binary.  So using `endsWith` for now